modules/custom/openy_activity_finder/js/programs_search.js

```diff
@@ -104,11 +126,26 @@
           }
           // If we need to add and it was not checked yet.
           if (key == -1 && !removeValue) {
-            Vue.set(this.checked, this.checked.length, this.dependencies[value][i]);
+            var setVal = this.dependencies[value][i];
+            if (!isNaN(setVal)) {
+              setVal = +setVal;
+            }
+            Vue.set(this.checked, this.checked.length, setVal);
           }
           // If already checked but we need to uncheck.
           if (key != -1 && removeValue) {
-            Vue.set(this.checked, key, '');
+            for (let k = 0; k < this.dependencies[value].length; k++) {
+              for (let j = 0; j < this.checked.length; j++) {
+                if (this.checked[j] == this.dependencies[value][k]) {
+                  this.checked.splice(this.checked.indexOf(this.dependencies[value][k]), 1);
+                }
+                if (this.checked[j] == this.dependencies[value][k].toString()) {
+                  this.checked.splice(this.checked.indexOf(this.dependencies[value][k].toString()), 1);
+                }
+              }
+            }
+            // This was default setter. Let's leave this as comment. @todo remove after qa.
```

Comment:
let's avoid of having commented code like this

Reply:
fixed